src/spikeinterface/core/result_core.py

```diff
@@ -0,0 +1,269 @@
+        some_spikes = spikes[self.sorting_result.random_spikes_indices]
+        
+        nbefore = int(self.params["ms_before"] * sorting.sampling_frequency / 1000.0)
+        nafter = int(self.params["ms_after"] * sorting.sampling_frequency / 1000.0)
+
+        if self.format == "binary_folder":
+            # in that case waveforms are extacted directly in files
+            file_path = self._get_binary_extension_folder() / "waveforms.npy"
+            mode = "memmap"
+            copy = False
+        else:
+            file_path = None
+            mode = "shared_memory"
+            copy = True
+
+        if self.sparsity is None:
+            sparsity_mask = None
+        else:
+            sparsity_mask = self.sparsity.mask
+
+        # TODO propagate some job_kwargs
+        job_kwargs = dict(n_jobs=-1)
+
+        all_waveforms = extract_waveforms_to_single_buffer(
+            recording,
+            some_spikes,
+            unit_ids,
+            nbefore,
+            nafter,
+            mode=mode,
+            return_scaled=self.params["return_scaled"],
+            file_path=file_path,
+            dtype=self.params["dtype"],
+            sparsity_mask=sparsity_mask,
+            copy=copy,
+            job_name="compute_waveforms",
+            **job_kwargs,
+        )
+
+        self.data["waveforms"] = all_waveforms
+
+    def _set_params(self, 
+            ms_before: float = 1.0,
+            ms_after: float = 2.0,
+            return_scaled: bool = True,
+            dtype=None,
+        ):
+        recording = self.sorting_result.recording
+        if dtype is None:
+            dtype = recording.get_dtype()
+
+        if return_scaled:
+            # check if has scaled values:
+            if not recording.has_scaled():
+                print("Setting 'return_scaled' to False")
+                return_scaled = False
+
+        if np.issubdtype(dtype, np.integer) and return_scaled:
+            dtype = "float32"
+
+        dtype = np.dtype(dtype)
+
+        params = dict(
+            ms_before=float(ms_before),
+            ms_after=float(ms_after),
+            return_scaled=return_scaled,
+            dtype=dtype.str,
+        )
+        return params
+
+    def _select_extension_data(self, unit_ids):
+        keep_unit_indices = np.flatnonzero(np.isin(self.sorting_result.unit_ids, unit_ids))
+        spikes = self.sorting_result.sorting.to_spike_vector()
+        some_spikes = spikes[self.sorting_result.random_spikes_indices]
+        keep_spike_mask = np.isin(some_spikes["unit_index"], keep_unit_indices)
+
+        new_data = dict()
+        new_data["waveforms"] = self.data["waveforms"][keep_spike_mask, :, :]
+
+        return new_data
+
+
+
+
+compute_waveforms = ComputeWaveforms.function_factory()
+register_result_extension(ComputeWaveforms)
+
+
+class ComputeTemplates(ResultExtension):
+    """
+    ResultExtension that compute templates (average, str, median, percentile, ...)
```

Comment:
```suggestion
    ResultExtension that computes templates (average, std, median, percentile, ...)
```

I assume str is std?